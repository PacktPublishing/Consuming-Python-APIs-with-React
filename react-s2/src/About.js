import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

class About extends Component {
  render() {
    return (
      <div className="App flex-container">
        <div className="About">
          <p>This is about page</p>
          <p><Link to="/">Go to the home page</Link></p>
        </div>
      </div>
    )
  }
}

export default About;
