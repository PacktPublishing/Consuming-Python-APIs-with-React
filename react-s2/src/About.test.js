import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";

// Set the default serializer for Jest to be the from enzyme-to-json
// This produces an easier to read (for humans) serialized format.
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import About from "./About.js";

test("renders correctly", () => {
  const wrapper = shallow(
    <About />
  );
  expect(wrapper).toMatchSnapshot();
});
