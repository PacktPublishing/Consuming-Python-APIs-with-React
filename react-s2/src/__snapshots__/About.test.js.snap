// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App flex-container"
>
  <div
    className="About"
  >
    <p>
      This is about page
    </p>
    <p>
      <Link
        replace={false}
        to="/"
      >
        Go to the home page
      </Link>
    </p>
  </div>
</div>
`;
